### ISSUE TITLE
Things to research:

### ISSUE THREAD

**Issue by xXHellgate25Xx** [VinhDevNguyen/Scene-Text-Project#1]
## Bài toán nhận diện syntax kiểu OCR
- [x] Kiểm tra Input
- [x] Search for Vietnamese data.
- [ ] OCR
- [ ] Word2vec
- [x] Vanishing Gradient and Exploding Gradient
- [ ] RNN
- [ ] Transformer (can it predict with this high accuracy: https://demo.computervision.com.vn/try-now?id=1)
- [ ] Long short-term memory (LSTM)

**Comment by VinhDevNguyen** [comment]
## Exploding Gradient
Vấn đề của Exploding Gradient có thể khắc phục được bằng cách sử dụng phương pháp [Gradient Clipping](https://towardsdatascience.com/what-is-gradient-clipping-b8e815cdfb48)
## Các bước thực hiện để trích xuất thông tin từ chứng minh thư
![step](https://hiepph.github.io/ocr/Structure.png)
Chúng ta có thể sử dụng thư viện [VietOCR](https://github.com/pbcquoc/vietocr) khá hay 
https://fpt.ai/technical-view-fvi-end-end-vietnamese-id-card-ocr
https://techinsight.com.vn/goc-nhin-cong-nghe-ve-fvi-he-thong-ocr-nhan-dien-chung-minh-thu-nhan-dan-viet-nam/
https://github.com/PaddlePaddle/PaddleOCR

**Comment by xXHellgate25Xx** [comment]
## OCR
Object Detection using Single Box Multishot Detection (SSD) used for Cropper and Detector: https://medium.com/featurepreneur/object-detection-using-single-shot-multibox-detection-ssd-and-opencvs-deep-neural-network-dnn-d983e9d52652

https://towardsdatascience.com/get-started-with-deep-learning-ocr-136ac645db1d

## Vanishing Gradient
Vấn đề của Vanishing Gradient có thể được giải quyết bằng:
- Dùng một hàm activation không khiến đạo hàm (gradient) bị nhỏ đi. Ví dụ như Relu.
- Dùng Residual Networks (ResNet): https://d2l.ai/chapter_convolutional-modern/resnet.html or https://towardsdatascience.com/building-a-resnet-in-keras-e8f1322a49ba
- Dùng batch normalization
